### PR TITLE
Don’t use arrays in `consume` functions

### PR DESCRIPTION
**Pull request by ahoppen**
Creating arrays for `consume(ifAny:)` and friends incurs memory allocations and reference counting costs. Stamp out copies of `consume(if:)` for up to three different kinds and start using `RawTokenKindSubset` if more kinds need to be checked.

This improves performance of parsing MovieSwiftUI by 5%.

rdar://103774645

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Tests/SwiftParserTest/AttributeTests.swift [hunk 1]
```diff
@@ -98,7 +98,7 @@ final class AttributeTests: XCTestCase {
       @1️⃣rethrows
       protocol P { }
       """,
-      substructure: Syntax(TokenSyntax.identifier("rethrows")),
+      substructure: Syntax(TokenSyntax.keyword(.rethrows)),
```
Comment (rintaro) [inline Tests/SwiftParserTest/AttributeTests.swift:101]: Just curious, what changed the behavior?

Comment (ahoppen) [reply Tests/SwiftParserTest/AttributeTests.swift:101]: Oh that was because I forgot to check `remapping` in `consume(if:)`. Fixed now.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by rintaro — APPROVED** [review]

**Pull request merged**